pkg/cmd/cli/cmd/newbuild.go

```diff
@@ -0,0 +1,111 @@
+
+This command will try to create a BuildConfig for your application using images, or code that has
+a public repository located on your system. It will lookup the images on the
+local Docker installation (if available), a Docker registry, or an OpenShift image stream.
+If you specify a source code URL, it will set up a build that takes your source code and converts
+it into an image that can run inside of a pod. Local source must be in a git repository that has
+remote repository that the OpenShift instance can see.
+
+Once the BuildConfig will be created you may need to run a build with 'start-build'.`
+
+	newBuildExample = `  // Create a BuildConfig based on the source code in the current git repository (with a public remote) and a Docker image
+  $ %[1]s new-build . --docker-image=repo/langimage
+
+  // Create a NodeJS application based on the provided [image]~[source code] combination
+  $ %[1]s new-build openshift/nodejs-010-centos7~https://bitbucket.com/user/nodejs-app
+
+  // Create an application from a remote repository using its beta2 branch
+  $ %[1]s new-build https://github.com/openshift/ruby-hello-world#beta2`
+)
+
+// NewCmdNewBuild implements the OpenShift cli new-build command
+func NewCmdNewBuild(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	_, typer := f.Object()
+	config := newcmd.NewAppConfig(typer)
+
+	cmd := &cobra.Command{
+		Use:     "new-build (IMAGE | IMAGESTREAM | TEMPLATE | PATH | URL ...)",
+		Short:   "Create a new BuildConfig",
+		Long:    newBuildLong,
+		Example: fmt.Sprintf(newBuildExample, fullName),
+		Run: func(c *cobra.Command, args []string) {
+			err := RunNewBuild(f, out, c, args, config)
+			if err == errExit {
+				os.Exit(1)
+			}
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmd.Flags().Var(&config.SourceRepositories, "code", "Source code to use to build this application.")
+	cmd.Flags().VarP(&config.ImageStreams, "image", "i", "Name of an OpenShift image stream to use in the app.")
+	cmd.Flags().Var(&config.DockerImages, "docker-image", "Name of a Docker image to include in the app.")
+	cmd.Flags().StringVar(&config.Name, "name", "", "Set name to use for generated application artifacts")
+	cmd.Flags().StringVar(&config.TypeOfBuild, "build", "", "Specify the type of build to use if you don't want to detect (docker|source).")
+	cmd.Flags().StringP("labels", "l", "", "Label to set in all resources for this application.")
+
+	cmd.Flags().StringP("output", "o", "", "Output format. One of: json|yaml.") //|template|templatefile.")
+	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
+	cmd.Flags().Bool("no-headers", false, "When using the default output, don't print headers.")
+	cmd.Flags().String("template", "", "Template string or path to template file to use when -o=template or -o=templatefile.  The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview]")
```

Comment:
We don't want to support real templates - you can go back to using the AddPrinterFlags function